aya-log-common/src/lib.rs

```diff
@@ -61,15 +64,35 @@ pub enum ArgType {
     F32,
     F64,
 
+    ArrU8Len16,
+    ArrU16Len8,
+
     Str,
 }
 
+/// All display hints
+#[repr(u8)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, IntoPrimitive)]
+pub enum DisplayHint {
+    /// Default string representation.
+    Default = 1,
+    /// `:x`
+    LowerHex,
+    /// `:X`
+    UpperHex,
+    /// `:ipv4`, `:IPv4`
```

Comment:
what is the reason for allowing both cases for ipv4? x and X don't work that way so seems inconsistent